web/frontend/src/components/gamepad.vue

```diff
@@ -37,131 +37,188 @@ const curStates = $ref<Record<string, number>>({
 let handle = -1;
 let prevStates: Record<string, number> = {};
 
+let lastError = Date.now();
 const sendEvent = (newEvent: InputController.Event) => {
-  if (enabled) {
-    const req = new InputController.TriggerEventRequest();
-    req.setController('WebGamepad');
-    req.setEvent(newEvent);
-    window.inputControllerService.triggerEvent(req, new grpc.Metadata(), displayError);
+  if (!enabled) {
+    return;
+  }
+  const req = new InputController.TriggerEventRequest();
+  req.setController('WebGamepad');
+  req.setEvent(newEvent);
+  window.inputControllerService.triggerEvent(req, new grpc.Metadata(), (error: ServiceError | null) => {
+    if (error) {
+      const now = Date.now();
+      if (now - lastError > 1000) {
+        lastError = now;
+        toast.error(error.message);
+      }
+    }
+  });
+};
+
+let lastTS = Timestamp.fromDate(new Date());
+const nextTS = () => {
```

Comment:
this annoying logic has us, with respect to this one client, actually send unique timestamps. Since JS Date will give us ms (without using perf api), many times we are clobbering our own events with the same timestamps. this results in a high chance, due to how server side processing works, of getting sticky controls.

That is, goroutines fire off per event and if two events with T=1 come in, you may choose the "wrong" event.

Reply:
Like you hinted, `performance.now()` has sub-ms precision. Just curious: why not use that here?

Reply:
Mmm nvm. Disregard my comment. Did not know that [because of security risks](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now) this method sometimes rounds to the ms.